components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb/src/org/wso2/integrationstudio/gmf/esb/HTTPEndpointOAuthAuthenticationMode.java

```diff
@@ -0,0 +1,219 @@
+    /**
+     * The '<em><b>Header OAuth Authentication Mode</b></em>' literal object.
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     * @see #HEADER_OAUTH_AUTHENTICATION_MODE_VALUE
+     * @generated
+     * @ordered
+     */
+    HEADER_OAUTH_AUTHENTICATION_MODE(0, "Header_OAuth_Authentication_Mode", "Header OAuth Authentication Mode"),
+
+    /**
+     * The '<em><b>Payload OAuth Authentication Mode</b></em>' literal object.
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     * @see #PAYLOAD_OAUTH_AUTHENTICATION_MODE_VALUE
+     * @generated
+     * @ordered
+     */
+    PAYLOAD_OAUTH_AUTHENTICATION_MODE(1, "Payload_OAuth_Authentication_Mode", "Payload OAuth Authentication Mode");
+
+    /**
+     * The '<em><b>Header OAuth Authentication Mode</b></em>' literal value.
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     * @see #HEADER_OAUTH_AUTHENTICATION_MODE
+     * @model name="Header_OAuth_Authentication_Mode" literal="Header OAuth Authentication Mode"
+     * @generated
+     * @ordered
+     */
+    public static final int HEADER_OAUTH_AUTHENTICATION_MODE_VALUE = 0;
+
+    /**
+     * The '<em><b>Payload OAuth Authentication Mode</b></em>' literal value.
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     * @see #PAYLOAD_OAUTH_AUTHENTICATION_MODE
+     * @model name="Payload_OAuth_Authentication_Mode" literal="Payload OAuth Authentication Mode"
+     * @generated
+     * @ordered
+     */
+    public static final int PAYLOAD_OAUTH_AUTHENTICATION_MODE_VALUE = 1;
+
+    /**
+     * An array of all the '<em><b>HTTP Endpoint OAuth Authentication Mode</b></em>' enumerators.
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     * @generated
+     */
+    private static final HTTPEndpointOAuthAuthenticationMode[] VALUES_ARRAY =
+        new HTTPEndpointOAuthAuthenticationMode[] {
+            HEADER_OAUTH_AUTHENTICATION_MODE,
+            PAYLOAD_OAUTH_AUTHENTICATION_MODE,
+        };
+
+    /**
+     * A public read-only list of all the '<em><b>HTTP Endpoint OAuth Authentication Mode</b></em>' enumerators.
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     * @generated
+     */
+    public static final List<HTTPEndpointOAuthAuthenticationMode> VALUES = Collections.unmodifiableList(Arrays.asList(VALUES_ARRAY));
+
+    /**
+     * Returns the '<em><b>HTTP Endpoint OAuth Authentication Mode</b></em>' literal with the specified literal value.
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     * @param literal the literal.
+     * @return the matching enumerator or <code>null</code>.
+     * @generated
+     */
+    public static HTTPEndpointOAuthAuthenticationMode get(String literal) {
+        for (int i = 0; i < VALUES_ARRAY.length; ++i) {
+            HTTPEndpointOAuthAuthenticationMode result = VALUES_ARRAY[i];
+            if (result.toString().equals(literal)) {
```

Comment:
Can result become null? If so shall we check for null

Reply:
_result_ cannot be null since _VALUES_ARRAY_ has initialized final at [line 82](https://github.com/wso2/integration-studio/pull/1063/files/015affafd1793a5a58aa3f2510ffd4461f465eae#:~:text=private%20static%20final,%7D%3B)